crypto/core_namemap.c

```diff
@@ -312,13 +322,22 @@ int ossl_namemap_add_names(OSSL_NAMEMAP *namemap, int number,
      */
     for (p = names; *p != '\0'; p = (q == NULL ? p + l : q + 1)) {
         int this_number;
+        char *allocated;
```

Comment:
Would doing a single strdup of names before the loop any better here?

Reply:
Probably.

Reply:
Good point, I'll do this.